apps/quick-learn-backend/src/routes/users/users.service.ts

```diff
@@ -141,6 +142,45 @@ export class UsersService extends PaginationService<UserEntity> {
     return users;
   }
 
+  async getUserRoadmaps(userId: number, includeCourses = false) {
+    const queryBuilder = this.userRepository
+      .createQueryBuilder('user')
+      .leftJoinAndSelect('user.assigned_roadmaps', 'roadmap')
+      .leftJoinAndSelect('roadmap.roadmap_category', 'roadmap_category')
+      .where('user.id = :userId', { userId })
+      .andWhere('roadmap.archived = :archived', { archived: false });
+
+    if (includeCourses) {
+      queryBuilder
+        .leftJoinAndSelect('roadmap.courses', 'course')
+        .andWhere('course.archived = :courseArchived', {
+          courseArchived: false,
+        });
+    }
+
+    const user = await queryBuilder.getOne();
+
+    if (!user?.assigned_roadmaps) {
+      return [];
+    }
+
+    // Map the results to include hardcoded percentage
```

Comment:
As discussed, we are not going to show the progress in this sprint and this should be always dynamic.